src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -0,0 +1,240 @@
+import { Request, RequestHandler } from 'express'
+
+import { createLoggerWithLabel } from '../../../../config/logger'
+import { AuthType, FieldResponse } from '../../../../types'
+import { CaptchaFactory } from '../../../services/captcha/captcha.factory'
+import { MyInfoFactory } from '../../../services/myinfo/myinfo.factory'
+import { createReqMeta } from '../../../utils/request'
+import * as FormService from '../../form/form.service'
+import { SpcpFactory } from '../../spcp/spcp.factory'
+import {
+  createCorppassParsedResponses,
+  createSingpassParsedResponses,
+} from '../../spcp/spcp.util'
+import * as SubmissionService from '../submission.service'
+
+import * as EmailSubmissionService from './email-submission.service'
+import {
+  mapAttachmentsFromResponses,
+  mapRouteError,
+} from './email-submission.util'
+
+const logger = createLoggerWithLabel(module)
+
+export const handleEmailSubmission: RequestHandler<
+  { formId: string },
+  { message: string; submissionId?: string; spcpSubmissionFailure?: boolean },
+  { responses: FieldResponse[]; isPreview: boolean },
+  { captchaResponse?: unknown }
+> = async (req, res) => {
+  const { formId } = req.params
+  const logMeta = {
+    action: 'handleEmailSubmission',
+    ...createReqMeta(req as Request),
+    formId,
+  }
+  // Retrieve form
+  const formResult = await FormService.retrieveFullFormById(
+    req.params.formId,
```

Comment:
FYI, injecting anything from `req` into a database query would usually raise red flags because of the potential for NoSQL injection. Although I believe we already rely on Express' route regex `a-fA-F0-9]{24}`, the implementation is decoupled. It might be better to move to [express-validator](https://express-validator.github.io/docs/index.html) pattern in the future so that the `RequestHandler` must be shipped together with validation code.

```suggestion
  const formResult = await FormService.retrieveFullFormById(
    formId,
```